core/src/test/java/org/apache/iceberg/view/ViewCatalogTests.java

```diff
@@ -1669,4 +1669,52 @@ public void concurrentReplaceViewVersion() {
                   .build());
     }
   }
+
+  @Test
+  public void testSqlForMultipleDialects() {
+    TableIdentifier identifier = TableIdentifier.of("ns", "view");
+
+    if (requiresNamespaceCreate()) {
+      catalog().createNamespace(identifier.namespace());
+    }
+
+    View view =
+        catalog()
+            .buildView(identifier)
+            .withSchema(SCHEMA)
+            .withDefaultNamespace(identifier.namespace())
+            .withDefaultCatalog(catalog().name())
+            .withQuery("spark", "select * from ns.tbl")
+            .withQuery("trino", "select * from ns.tbl using X")
+            .create();
+
+    assertThat(view.sqlFor("spark").sql()).isEqualTo("select * from ns.tbl");
+    assertThat(view.sqlFor("trino").sql()).isEqualTo("select * from ns.tbl using X");
+    assertThat(view.sqlFor("unknown-dialect").sql()).isEqualTo("select * from ns.tbl");
+  }
+
+  @Test
+  public void testSqlForInvalidArguments() {
+    TableIdentifier identifier = TableIdentifier.of("ns", "view");
+
+    if (requiresNamespaceCreate()) {
+      catalog().createNamespace(identifier.namespace());
+    }
+
+    View view =
```

Comment:
It looks like we can rely on catalog cleanup because the different extensions AfterEach will close (which clears the state from the catalog)